src/plugins/vis_builder/public/application/utils/validations/vis_builder_state_validation.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import Ajv from 'ajv';
+import visBuilderStateSchema from '../schema.json';
+import { ValidationResult } from './types';
+
+const ajv = new Ajv();
+const validateState = ajv.compile(visBuilderStateSchema);
+
+export const validateVisBuilderState = (visBuilderState: any): ValidationResult => {
```

Comment:
This is the visBuilder state loaded from saved objects. I kept the type signature as `any` because the input to this function could in theory be very different from the `RootState` interface and we are essentially validating here that it is the expected type

Reply:
ah, I got it - didn't realize we were dealing with the raw saved object state here.